### DISCUSSION TITLE
[Miscellaneous] Regrading OP-Geth Performance and improvement

### DISCUSSION THREAD

**Discussion by hbisht-airstack** [ethereum-optimism/developers#82] [Miscellaneous]
### Issue Description

Hi There,
I am new to op-geth, and I am using it with base, but the node on docker isn't able to sustain a load of 20rps and throwing lot of errors for calls such as get balance/symbols even with 32 cores, 128gb and nvme ssd servers.

**Also while in the logs we are getting execution aborted (timeout = 15s) the CPU utilization is 100% on the server.**
<img width="1511" alt="Screenshot 2023-11-22 at 2 12 41 PM" src="https://github.com/ethereum-optimism/developers/assets/42200799/7fc51f9c-d01a-4a88-966d-a098ea66b1a7">

**Logs on server**
`WARN [11-21|16:49:02.373] Served eth_call                          conn=34.200.220.241:34282 reqid=1 duration=18.714470318s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.373] Served eth_call                          conn=34.200.220.241:38723 reqid=1 duration=19.373425659s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.373] Served eth_call                          conn=34.200.220.241:22217 reqid=1 duration=19.362657833s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.373] Served eth_call                          conn=34.200.220.241:12412 reqid=1 duration=16.710411102s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.373] Served eth_call                          conn=34.200.220.241:33072 reqid=1 duration=18.653971826s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.374] Served eth_call                          conn=34.200.220.241:52496 reqid=1 duration=15.032813245s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.374] Served eth_call                          conn=34.200.220.241:26695 reqid=1 duration=19.038112161s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.374] Served eth_call                          conn=34.200.220.241:29175 reqid=1 duration=18.661341256s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.375] Served eth_call                          conn=34.200.220.241:15183 reqid=1 duration=18.722897473s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.375] Served eth_call                          conn=34.200.220.241:55664 reqid=1 duration=17.131542127s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.375] Served eth_call                          conn=34.200.220.241:11816 reqid=1 duration=19.197091788s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.375] Served eth_call                          conn=34.200.220.241:28009 reqid=1 duration=18.733107418s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.376] Served eth_call                          conn=34.200.220.241:10663 reqid=1 duration=19.224301574s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.376] Served eth_call                          conn=34.200.220.241:7449  reqid=1 duration=16.605749498s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.377] Served eth_call                          conn=34.200.220.241:49755 reqid=1 duration=16.713432895s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.377] Served eth_call                          conn=34.200.220.241:19077 reqid=1 duration=17.17344042s  err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.378] Served eth_call                          conn=34.200.220.241:43671 reqid=1 duration=19.229840376s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.378] Served eth_call                          conn=34.200.220.241:45859 reqid=1 duration=19.260660233s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.378] Served eth_call                          conn=34.200.220.241:63756 reqid=1 duration=19.284244109s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:61026 reqid=1 duration=16.737384468s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:3552  reqid=1 duration=16.608951097s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:8220  reqid=1 duration=19.202474879s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:57308 reqid=1 duration=19.363375543s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:49723 reqid=1 duration=18.747517394s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.098] Served eth_call                          conn=34.200.220.241:50208 reqid=1 duration=18.444512874s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:12330 reqid=1 duration=18.729891783s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.379] Served eth_call                          conn=34.200.220.241:26225 reqid=1 duration=16.620415422s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:12408 reqid=1 duration=16.703205448s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:26279 reqid=1 duration=19.374863062s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:47163 reqid=1 duration=16.73895495s  err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.102] Served eth_call                          conn=34.200.220.241:44138 reqid=1 duration=18.455994402s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:9928  reqid=1 duration=19.273609672s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:9267  reqid=1 duration=19.243257158s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.121] Served eth_call                          conn=34.200.220.241:32363 reqid=1 duration=19.030445857s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:40754 reqid=1 duration=16.699951082s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.380] Served eth_call                          conn=34.200.220.241:22357 reqid=1 duration=16.741313884s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.163] Served eth_call                          conn=34.200.220.241:43491 reqid=1 duration=19.077995022s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.381] Served eth_call                          conn=34.200.220.241:41910 reqid=1 duration=16.670019032s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.231] Served eth_call                          conn=34.200.220.241:1760  reqid=1 duration=17.521040628s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.381] Served eth_call                          conn=34.200.220.241:19738 reqid=1 duration=18.943232282s err="execution aborted (timeout = 15s)"
WARN [11-21|16:49:02.381] Served eth_call                          conn=34.200.220.241:18305 reqid=1 duration=16.610956341s err="execution aborted (timeout = 15s)"`

**Details of OP-Geth**
```
REPO=https://github.com/ethereum-optimism/op-geth.git
VERSION=v1.101304.1
COMMIT=ed8e9f54ed0a5069e77ea34895ff12a9ac4ca17f
```
What should be the fix for this.

**Comment by sbvegan** [comment]
Hey y'all, I also responded in telegram. I spoke to a couple client engineers about this. They noted that this is an older block (~90days), so you should use an archive node to query this historical state.

What's probably happening is your full node is trying to reconstruct the historical state and is timing out because it takes a long time to do that.

It was also noted that this block happened during the friend.tech craze. At the time there were a lot of MEV bots trying to snipe shares/keys so there are a lot of revert events emitted that could be adding additional latency to the block fetching.

**Comment by sbvegan** [comment]
Closing this because it was resolved out of this thread. The issue was a due to a backend RPC request misconfiguration.
